src/Nethermind/Nethermind.Synchronization/FastSync/TreeSync.cs

```diff
@@ -334,6 +334,7 @@ void AddAgainAllItems()
                 }
                 finally
                 {
+                    batch.Dispose();
```

Comment:
`.Dispose` can't fail can it; or it wouldn't release the lock. Maybe switch these two, releasing the lock earlier is better anyway.

Reply:
Well only thing it does is Pool.Return